### PR TITLE
Add custom ssh key setup to fix actions

### PR DESCRIPTION
**Pull request by J-Thompson12**
**What changed?**
Copied over the ssh known_hosts fix from https://github.com/weaveworks/weave-gitops-enterprise/pull/312. This fix is in response to https://github.blog/2021-09-01-improving-git-protocol-security-github/. 

**How did you test it?**
If the pipeline passes its good.

### CODE REVIEW

**Comment by foot** [comment]
So it looks like `test` also needs the known_hosts adjustment. We copied the whole:

```
- name: Set up ssh
    uses: ./.github/actions/setup-ssh
    with:
      ssh-private-key: ${{ inputs.github-key }}
```

Into one of our tests, even though it doesn't need the private key. 🤷 . If thats not an option you could untangle the ssh-agent and known hosts bits.

**Review by jpellizzari — APPROVED** [review]
LGTM if the pipeline passes

**Comment by foot** [comment]
Hmmm, its a bit weird, looks like its stuck.. 👀

**Comment by J-Thompson12** [comment]
> Hmmm, its a bit weird, looks like its stuck.. 👀

Yeah I was just looking at that but couldnt see why. I just restarted the tests to see if it happens again.

File: .github/actions/setup-ssh/action.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+description: 'Setup SSH keys and known hosts'
+inputs:
+  ssh-private-key:
+    description: 'ssh keys to load'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Set up ssh
+      uses: ./.github/actions/setup-ssh
```
Comment (foot) [inline .github/actions/setup-ssh/action.yml:12]: ♾️ !

Comment (foot) [inline .github/actions/setup-ssh/action.yml:12]:
```suggestion
    - name: Set up ssh agent
      uses: webfactory/ssh-agent@v0.5.3
```

Comment (J-Thompson12) [reply .github/actions/setup-ssh/action.yml]: Ohhh nice catch. I guess I did just find all. Whoops

**Comment by foot** [comment]
Copied the wge one over to see if whitespace issues..

**Pull request merged**